compiler/x/codegen/OMRAheadOfTimeCompile.cpp

```diff
@@ -0,0 +1,142 @@
+ *******************************************************************************/
+
+#include "codegen/AheadOfTimeCompile.hpp"
+#include "env/FrontEnd.hpp"
+#include "codegen/Instruction.hpp"
+#include "compile/Compilation.hpp"
+#include "compile/ResolvedMethod.hpp"
+#include "runtime/TRRelocationRecord.hpp"
+#include "compile/VirtualGuard.hpp"
+#include "env/CompilerEnv.hpp"
+#include "env/jittypes.h"
+#include "il/Node.hpp"
+#include "il/Node_inlines.hpp"
+#include "il/SymbolReference.hpp"
+#include "ras/DebugCounter.hpp"
+#include "runtime/CodeCacheConfig.hpp"
+#include "runtime/CodeCacheManager.hpp"
+
+#ifdef OMR_RELOCATION_RUNTIME
+
+TR::AheadOfTimeCompile*
+OMR::X86::AheadOfTimeCompile::self()
+   {
+   return static_cast<TR::AheadOfTimeCompile *> (this);
+   }
+
+void OMR::X86::AheadOfTimeCompile::processRelocations()
+   {
+   TR::Compilation* comp = self()->comp();
+   TR_FrontEnd *fe = comp->fe();
+   TR::CodeGenerator* cg = comp->cg();
+   // calculate the amount of memory needed to hold the relocation data
+
+   for (auto aotIterator = cg->getExternalRelocationList().begin(); aotIterator != cg->getExternalRelocationList().end(); ++aotIterator)
+      (*aotIterator)->addExternalRelocation(cg);
+
+   TR::IteratedExternalRelocation *r;
+   for (r = self()->getAOTRelocationTargets().getFirst();
+        r != NULL;
+        r = r->getNext())
+      {
+      self()->addToSizeOfAOTRelocations(r->getSizeOfRelocationData());
+      }
+
+   // now allocate the memory  size of all iterated relocations + the header (total length field)
+
+   // Note that when using the SymbolValidationManager, the well-known classes
+   // must be checked even if no explicit records were generated, since they
+   // might be responsible for the lack of records.
+
+   if (self()->getSizeOfAOTRelocations() != 0)
+      {
+      // It would be more straightforward to put the well-known classes offset
+      // in the AOT method header, but that would use space for AOT bodies that
+      // don't use the SVM. TODO: Move it once SVM takes over?
+      uintptr_t reloBufferSize =
+         self()->getSizeOfAOTRelocations() +OMR::AheadOfTimeCompile::SIZEPOINTER ;
+      uint8_t *relocationDataCursor = self()->setRelocationData(
+         fe->allocateRelocationData(self()->comp(), reloBufferSize));
+      // set up the size for the region
+      *(uintptr_t *)relocationDataCursor = reloBufferSize;
+      relocationDataCursor +=  OMR::AheadOfTimeCompile::SIZEPOINTER;
+      }
+    if (self()->getSizeOfAOTRelocations() != 0)
+      {
+      uint8_t *relocationDataCursor = self()->setRelocationData(fe->allocateRelocationData(self()->comp(), self()->getSizeOfAOTRelocations() + sizeof(uintptr_t )));
+      uint8_t * copy = relocationDataCursor;
+      // set up the size for the region
+      TR::RelocationRecordBinaryTemplate* groups = reinterpret_cast<
+      TR::RelocationRecordBinaryTemplate*> (relocationDataCursor);
```

Comment:
Splitting the line this way is *really* confusing. Please don't line up the cast type like it's a variable definition.

Reply:
Oh, that's done by mistake. Fixed that now, thank you for the comment.